src/method-dispatch.c

```diff
@@ -145,47 +183,68 @@ SEXP method_call_(SEXP call, SEXP generic, SEXP envir) {
   ++n_protect;
   SEXP mcall_tail = mcall;
 
+  PROTECT_INDEX arg_pi;
+  PROTECT_WITH_INDEX(R_NilValue, &arg_pi);
+  ++n_protect;
+
   // For each of the arguments to the generic
   for (R_xlen_t i = 0; i < n_args; ++i) {
 
     // Find its name and look up its value (a promise)
     SEXP name = TAG(formals);
-    SEXP arg = Rf_findVar(name, envir);
+    SEXP arg = Rf_findVarInFrame(envir, name);
 
+    // n_dispatch is always either 1 or 2.
```

Comment:
Can't it be any number?

Reply:
Hmm, when writing the R-dev blog I remember an edit made (I forget by who) adding/correcting this line: 

> while S4 generics can dispatch on an arbitrary number of arguments. S7 implements double dispatch (i.e. multiple dispatch on exactly two arguments) 

but testing locally just now it seems we *do* support dispatch on more than one arg. 

Thanks for the correction!